xarray/core/indexing.py

```diff
@@ -173,8 +173,9 @@ def convert_label_indexer(index, label, index_name="", method=None, tolerance=No
             else _asarray_tuplesafe(label)
```

Comment:
Maybe we could avoid casting to ndarray on this line? All we really need it for is checking dimensionality of the result.

Reply:
we're also using `label.dtype.kind`, so I'm not sure how we can avoid casting to ndarray?

Reply:
I guess we could simply extract `label.ndim` and `label.dtype`?